src/arch/timer.h

```diff
@@ -71,22 +72,36 @@ static inline timer_uint timer_value(timer_uint start)
 	return (timer_new() - start) * 1000000U / timer_perf_freq;
 }
 
+#else
+
+static inline timer_uint timer_new(void)
+{
+	return 0;
+}
+
+static inline timer_uint timer_value(timer_uint start)
+{
+	return 0;
+}
+
```

Comment:
Correct, it will never compile. This is just an ugly workaround so that Doxygen can see the existence of those functions; as you can see [here](https://root-sim.github.io/core/docs/develop/timer_8h.html), the documentation for the timers was broken.
Anyway, I agree, this looks horrible. I found another cleaner workaround that doesn't require bogus definitions: simply add function declarations for the conditionally compiled code and slap the Doxygen documentation on that. I will implement this approach in the next commit, so you can see if you like the solution.

Reply:
I've seen the workaround and it looks great!